src/v/test_utils/runfiles.cc

```diff
@@ -17,8 +17,13 @@ get_runfile_path([[maybe_unused]] std::string_view path) {
 #ifdef BAZEL_TEST
     using bazel::tools::cpp::runfiles::Runfiles;
     std::string error;
-    std::unique_ptr<Runfiles> runfiles(
-      Runfiles::CreateForTest(BAZEL_CURRENT_REPOSITORY, &error));
+    std::unique_ptr<Runfiles> runfiles;
+    if (std::getenv("BAZEL_TEST") != nullptr) {
+        runfiles.reset(
+          Runfiles::CreateForTest(BAZEL_CURRENT_REPOSITORY, &error));
```

Comment:
So test and non-test have different runfiles layout, and that's why you need call these two methods, right?

I'm a bit confused by `Runfiles` library itself doens't have this method which detects if it's in a test: isn't that the most common code: when you write code it could be invoked in a test or not (with the exception of code in _test.cc type files itself), so you'd need to do the switching thing.

Reply:
> So test and non-test have different runfiles layout, and that's why you need call these two methods, right?

Not different layout, but it's a portability thing. In tests Bazel is always invoking and setting the correct environment variables to lookup the correct runfiles. For binaries, it looks up different environment variables or looks for the runfiles based on the executable location (which we don't support and pass empty string as the docs recommend). This is why I think there are both APIs available.

Yes I agree and have wondered this myself...

Reply:
> For binaries, it looks up different environment variables or looks for the runfiles based on the executable location (which we don't support and pass empty string as the docs recommend)

Does the "which we don't support", does that mean these benchmarks are going to fail when run raw like `bazel-bin/.../foo_rpbench` ? Is it easy to support this?

Reply:
Well I think it might work, but not for the crypto bench because it requires other special handling like the openssl config file that needs to have the search path updated. I wonder if `--run_under` can solve some of these use cases.

I am doing some testing, of this. I don't think the openssl ones will work very easily (because of the OpenSSL search path business), although the wasm ones should.

Reply:
Let me put up a followup PR for now

Reply:
Okay just confirmed the wasm ones work fine with directly invoking the binary. It's just the openssl ones that don't.

Reply:
I think it will take extra work to get the openssl ones to "just work"

> Okay just confirmed the wasm ones work fine with directly invoking the binary.

And this is with a small patch I'll send up as a followup